src/lib/statistics/table_statistics.cpp

```diff
@@ -33,35 +54,56 @@ std::shared_ptr<TableStatistics> TableStatistics::from_table(const Table& table)
     threads.emplace_back([&] {
       while (true) {
         auto my_column_id = ColumnID{static_cast<ColumnID::base_type>(next_column_id++)};
-        if (static_cast<ColumnCount>(my_column_id) >= table.column_count()) return;
+        if (static_cast<ColumnCount>(my_column_id) >= table.column_count()) {
+          return;
+        }
 
         const auto column_data_type = table.column_data_type(my_column_id);
 
         resolve_data_type(column_data_type, [&](auto type) {
           using ColumnDataType = typename decltype(type)::type;
 
           const auto output_column_statistics = std::make_shared<AttributeStatistics<ColumnDataType>>();
-
-          const auto histogram =
-              EqualDistinctCountHistogram<ColumnDataType>::from_column(table, my_column_id, histogram_bin_count);
-
-          if (histogram) {
-            output_column_statistics->set_statistics_object(histogram);
-
-            // Use the insight that the histogram will only contain non-null values to generate the NullValueRatio
-            // property
-            const auto null_value_ratio =
-                table.row_count() == 0
-                    ? 0.0f
-                    : 1.0f - (static_cast<float>(histogram->total_count()) / static_cast<float>(table.row_count()));
-            output_column_statistics->set_statistics_object(
-                std::make_shared<NullValueRatioStatistics>(null_value_ratio));
+          const auto chunk_count = table.chunk_count();
+          const auto should_generate_small_histograms = (chunk_count > 1) && (column_data_type != DataType::String);
```

Comment:
I guess that's the "heuristic" to determine when to use segment-based histograms and when not?
If so, do you know how well it works?

Reply:
This basically says: use it whenever we don't have a string. So it's not really a heuristic.

Reply:
Does that you would use the "new approach" in all cases but strings?

Reply:
Yes, we do it in all cases but strings, but no, its not always a good idea.